src/QMCWaveFunctions/TWFPrototype.h

```diff
@@ -0,0 +1,229 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2021 QMCPACK developers.
+//
+// File developed by:   Raymond Clay III, rclay@sandia.gov, Sandia National Laboratories
+//
+// File created by:   Raymond Clay III, rclay@sandia.gov, Sandia National Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_TWFPROTOTYPE_H
+#define QMCPLUSPLUS_TWFPROTOTYPE_H
+
+#include "QMCWaveFunctions/WaveFunctionComponent.h"
+#include "QMCWaveFunctions/SPOSet.h"
+#include "Configuration.h"
+#include "Particle/ParticleSet.h"
+namespace qmcplusplus
+{
+/**
+ *  TWFPrototype is a wrapper class for TrialWavefunction that provides separate and low level access to the Jastrow and 
+ *  SPOSet objects.  This is so that observables can be recast in matrix form and their derivatives taken efficiently. 
+ *  Currently this is hard coded for ground state slater jastrow wave functions, but generalization to
+ *  arbitrary occupations and multideterminants are straightforward and will come online as the code is tested and validated. 
+ *
+ *  Please see : J. Chem. Phys. 144, 194105 (2016) https://doi.org/10.1063/1.4948778 for implementation details and formalism.
+ */
+class TWFPrototype
```

Comment:
WaveFunctionForceWrapper?
TWFPrototype especially when you name an instance of it twf (used elsewhere in the code for trial wave function) isn't helping.
If we are just refering to the stage of development ForcesPrototype would be better.